test/e2e/framework/plugin_operations.go

```diff
@@ -13,9 +13,66 @@ import (
 	"gopkg.in/yaml.v3"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
-	cliv1alpha1 "github.com/vmware-tanzu/tanzu-cli/apis/cli/v1alpha1"
+	configtypes "github.com/vmware-tanzu/tanzu-plugin-runtime/config/types"
 )
 
+// ArtifactList contains an Artifact object for every supported platform of a version.
+type ArtifactList []Artifact
+
+// Artifact points to an individual plugin binary specific to a version and platform.
+type Artifact struct {
+	// Image is a fully qualified OCI image for the plugin binary.
+	Image string `json:"image,omitempty"`
+	// AssetURI is a URI of the plugin binary. This can be a fully qualified HTTP path or a local path.
+	URI string `json:"uri,omitempty"`
+	// SHA256 hash of the plugin binary.
+	Digest string `json:"digest,omitempty"`
+	// Type of the binary artifact. Valid values are S3, GCP, OCIImage.
+	Type string `json:"type"`
+	// OS of the plugin binary in `GOOS` format.
+	OS string `json:"os"`
+	// Arch is CPU architecture of the plugin binary in `GOARCH` format.
+	Arch string `json:"arch"`
+}
+
+// CLIPluginSpec defines the desired state of CLIPlugin.
+type CLIPluginSpec struct {
+	// Description is the plugin's description.
+	Description string `json:"description"`
+	// Recommended version that Tanzu CLI should use if available.
+	// The value should be a valid semantic version as defined in
+	// https://semver.org/. E.g., 2.0.1
+	RecommendedVersion string `json:"recommendedVersion"`
+	// Artifacts contains an artifact list for every supported version.
+	Artifacts map[string]ArtifactList `json:"artifacts,omitempty"`
+	// Optional specifies whether the plugin is mandatory or optional
+	// If optional, the plugin will not get auto-downloaded as part of
+	// `tanzu login` or `tanzu plugin sync` command
+	// To view the list of plugin, user can use `tanzu plugin list` and
+	// to download a specific plugin run, `tanzu plugin install <plugin-name>`
+	Optional bool `json:"optional,omitempty"`
+	// Target specifies the target of the plugin. Only needed for standalone plugins
+	Target configtypes.Target `json:"target,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// CLIPlugin denotes a Tanzu cli plugin.
+type CLIPlugin struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              CLIPluginSpec `json:"spec"`
+}
+
+//+kubebuilder:object:root=true
+
+// CLIPluginList contains a list of CLIPlugin
+type CLIPluginList struct {
```

Comment:
This one does not seem to be used.

Reply:
thanks for catching this. removed the unused struct